src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1610,6 +1605,11 @@ private static SyntaxTokenList AsModifierList(Accessibility accessibility, Decla
                 list = list.Add(SyntaxFactory.Token(SyntaxKind.StaticKeyword));
             }
 
+            if (modifiers.IsNew)
```

Comment:
💭  The containing method should be using the order specified by `CSharpCodeStyleOptions.PreferredModifierOrder`. I'm not sure we want to make a change to this implementation until we are ready to update it to always follow that option.